library/src/main/java/com/datatorrent/lib/parser/Parser.java

```diff
@@ -61,25 +70,48 @@ public void setup(PortContext context)
     }
   };
 
-  @OutputPortFieldAnnotation(optional = true)
   public transient DefaultOutputPort<INPUT> err = new DefaultOutputPort<INPUT>();
```

Comment:
As part of requirement, Csv parser operator , json and xml parser  would have to be enhanced to emit error tuples with error message . I get @chandnisingh suggestion of not keeping error port as KeyValuePair by default, but also its type shouldn't be INPUT as well which is the case now.   Better approach would be to have generic ERROR and leave the implementation to child operators